lib/wcc/contentful/store/lazy_cache_store.rb

```diff
@@ -2,25 +2,75 @@
 
 module WCC::Contentful::Store
   class LazyCacheStore
-    attr_reader :cache
+    delegate :find_all, to: :@cdn
 
-    delegate :find_all, to: :@store
+    # TODO: https://zube.io/watermarkchurch/development/c/2265
+    #  figure out how to cache the results of a find_by query, ex:
+    #  `find_by('slug' => '/about')`
+    delegate :find_by, to: :@cdn
 
-    def initialize(client:, cache: nil)
-      @store = CDNAdapter.new(client)
+    def initialize(client, cache: nil)
+      @cdn = CDNAdapter.new(client)
       @cache = cache || ActiveSupport::Cache::MemoryStore.new
     end
 
     def find(key)
-      @cache.fetch(key) do
-        @store.find(key)
+      found =
+        @cache.fetch(key) do
+          # if it's not a contentful ID don't hit the API.
+          # Store a nil object if we can't find the object on the CDN.
+          (@cdn.find(key) || nil_obj(key)) if key =~ /^\w{22}$/
```

Comment:
this always the format of a key?

Reply:
Yep all Contentful IDs are 22 character alphanumeric.  Though I definitely could understand if we don't want to build on that assumption...